src/message.c

```diff
@@ -2392,11 +2392,13 @@ qd_message_t *qd_message_compose(qd_composed_field_t *f1,
         qd_compose_free(fields[idx]);
     }
 
+    LOCK(&content->lock);
     // initialize the Q2 flag. Note that we do not need to hold the content
```

Comment:
The comment is out-of-date, now?

Reply:
I don't think this comment is out-of-date. I agree with the comments and am inclined to remove the lock I added. @kgiusti can you please confirm ?

Reply:
Yeah this lock shouldn't be necessary IIUC because the message hasn't yet been shared with any other thread - it was just created in this function and we're just setting the q2_input_holdoff initial value.

Fixing the tsan warning was not the purpose of the original patch, which is to simply corrected a missing atomic operation. I'd recommend committing the original one-line change only without removing the tsan suppression since IIUC the suppression is still necessary after the one-line change is made.

Reply:
I believe that the suppression can be removed with this change. I will take out the lock that I added (since it is not necessary) and take out the suppression as well and see if this passed CI. if it does, the suppression can be safely removed

Reply:
The CI has passed. I am merging this PR to main, thanks.